cartography/intel/gsuite/__init__.py

```diff
@@ -62,23 +61,50 @@ def start_gsuite_ingestion(neo4j_session: neo4j.Session, config: Config) -> None
-        credentials = credentials.create_delegated(GSUITE_DELEGATED_ADMIN)
-
-    except ApplicationDefaultCredentialsError as e:
-        logger.debug('Error occurred calling GoogleCredentials.get_application_default().', exc_info=True)
-        logger.error(
-            (
-                "Unable to initialize GSuite creds. If you don't have GSuite data or don't want to load "
-                'Gsuite data then you can ignore this message. Otherwise, the error code is: %s '
-                'Make sure your GSuite credentials are configured correctly, your credentials file (if any) is valid. '
-                'For more details see README'
-            ),
-            e,
-        )
-        return
+    if config.gsuite_auth_method == 'delegated':  #  Legacy delegated method
+        try:
+            credentials = GoogleCredentials.from_stream(config.gsuite_config)
+            credentials = credentials.create_scoped(OAUTH_SCOPE)
+            credentials = credentials.create_delegated(os.environ.get('GSUITE_DELEGATED_ADMIN'))
+
+        except ApplicationDefaultCredentialsError as e:
+            logger.debug('Error occurred calling GoogleCredentials.get_application_default().', exc_info=True)
+            logger.error(
+                (
+                    "Unable to initialize GSuite creds. If you don't have GSuite data or don't want to load "
+                    'Gsuite data then you can ignore this message. Otherwise, the error code is: %s '
+                    'Make sure your GSuite credentials file (if any) is valid. '
+                    'For more details see README'
+                ),
+                e,
+            )
+            return
+    elif config.gsuite_auth_method == 'oauth':
+        auth_tokens = json.loads(str(base64.b64decode(config.gsuite_config).decode()))
+        logger.info('SO GOOD SO FAR')
+        try:
+            credentials = GoogleCredentials(
+                None,
+                auth_tokens['client_id'],
+                auth_tokens['client_secret'],
+                auth_tokens['refresh_token'],
+                None,
+                auth_tokens['token_uri'],
+                'Cartography',
+            )
+            credentials.refresh(httplib2.Http())
+            credentials = credentials.create_scoped(OAUTH_SCOPE)
+        except ApplicationDefaultCredentialsError as e:
+            logger.debug('Error occurred calling GoogleCredentials.get_application_default().', exc_info=True)
+            logger.error(
+                (
+                    "Unable to initialize GSuite creds. If you don't have GSuite data or don't want to load "
+                    'Gsuite data then you can ignore this message. Otherwise, the error code is: %s '
+                    'Make sure your GSuite credentials are configured correctly, your credentials are valid. '
+                    'For more details see README'
+                ),
+                e,
+            )
```

Comment:
```suggestion
            )
            return
```

I think we want to return out of this function if this exception is hit, which will then continue on to run the other syncs.